src/wp-admin/includes/image-edit.php

```diff
@@ -391,6 +397,18 @@ function wp_save_image_file( $filename, $image, $mime_type, $post_id ) {
 				return imagepng( $image, $filename );
 			case 'image/gif':
 				return imagegif( $image, $filename );
+			case 'image/webp':
+				if ( function_exists( 'imagewebp' ) ) {
+					/**
+					 * Filters the WebP compression quality for image file saves.
+					 *
+					 * @since 5.8.0
+					 *
+					 * @param int    $quality Quality level between 0 (low) and 100 (high) of the WebP.
+					 */
+					return imagewebp( $image, null, apply_filters( 'webp_quality', 75 ) );
```

Comment:
Thanks for reviewing!

We discussed this a bit previously - 90 is for transforms, 75 for saves - see this and following comment - https://core.trac.wordpress.org/ticket/35725#comment:137

More research is probably warranted to further fine tune the default WebP quality (and can continue after the initial commit).

Reply:
This also seems a bit odd to me. If we are using different quality values here, why don't we do the same for JPEG? I agree more research here would be worth doing.

One other point of feedback that's more code-related: The `jpeg_quality` filter also passes additional context of `edit_image`. I'm thinking we should probably architect the new `webp_quality` filter in the same way and pass `edit_image` here as well.

Reply:
Added the 'edit_image' context.  Odd that is the context because this is actually the 'save_image' function, but we can keep that way to match jpeg.

As for the quality, although it is shown as 90 here for jpeg, the quality for saves is actually set to 82 here -> https://github.com/WordPress/wordpress-develop/blob/2382765afa36e10bf3c74420024ad4e85763a47c/src/wp-includes/class-wp-image-editor.php#L20 

I will double check that the quality settings are used as I expect for jpeg and WebP.

Reply:
Turns out these sections are legacy code and only apply when the passed image isn't an instance of `WP_Image_Editor`. I'm actually not sure that can happen, in my testing the image is always a `WP_Image_Editor` and the quality used is the default: 82. @getsource - any idea how an image could get passed that isn't an instance of `WP_Image_Editor`? If this is purely legacy code, we could remove the WebP support parts.

If we want to use a different value for WebP quality we could add this via the `wp_editor_set_quality` filter which passed the mime type as context. For now, feels fine to leave at the default value since we haven't studied the best level to use yet.